Dockerfile

```
@@ -1,27 +1,24 @@
-FROM ubuntu:22.04
+FROM ubuntu:jammy
 ENV DEBIAN_FRONTEND noninteractive
 
 RUN apt-get update
 RUN apt-get install --no-install-recommends -y apt-transport-https
 RUN apt-get update
 RUN apt-get install --no-install-recommends -y curl gnupg2 ca-certificates software-properties-common nlohmann-json3-dev
 
-RUN apt-get install --no-install-recommends -y git cmake build-essential sqlite3 libsqlite3-dev libssl-dev librdkafka-dev libboost-all-dev libtool libxerces-c-dev libflatbuffers-dev
+RUN apt-get install --no-install-recommends -y git cmake build-essential sqlite3 libsqlite3-dev libssl-dev librdkafka-dev libboost-all-dev libtool libxerces-c-dev libflatbuffers-dev libjsoncpp-dev libspdlog-dev pigz
 
 RUN add-apt-repository ppa:deadsnakes/ppa
-RUN apt-get install --no-install-recommends -y python3.11-dev
-RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
-RUN apt-get install --no-install-recommends -y libjsoncpp-dev libspdlog-dev pigz
-RUN python3.11 get-pip.py
-RUN pip install tensorflow==2.13.0
-RUN pip install -U scikit-learn
-RUN pip install joblib
-RUN pip install threadpoolctl
+RUN apt-get install --no-install-recommends -y python3.8-dev python3-pip python3.8-distutils
+RUN python3.8 -m pip install stellargraph
```

```suggestion
RUN pip install stellargraph
```
The `python3.8 -m` is needed when there are multiple python versions and to install the package specifically to the version specified

---

Yes, there is a Python 3.10 installation too (which is probably installed as a dependency of some other package). Also /usr/bin/python3 is linked to Python 3.10 (i.e. 3.10 is considered the default python3). So we need to explicitly specify python3.8 for pip install.